Source/WebCore/Modules/speech/SpeechRecognition.cpp

```diff
@@ -77,6 +78,11 @@ ExceptionOr<void> SpeechRecognition::startRecognition()
         return Exception { UnknownError, "Recognition is not in a valid frame"_s };
 
     auto optionalFrameIdentifier = document->frameID();
+    if (!isFeaturePolicyAllowedByDocumentAndAllOwners(FeaturePolicy::Type::Microphone, document.get(), LogFeaturePolicyFailure::No)) {
+        didError({ SpeechRecognitionErrorType::NotAllowed, "Permission is denied"_s });
```

Comment:
this seems weird to me, shouldn't startRecognition throw instead? 

What is chrome doing?

Reply:
Firing an event is more backward compatible (similar to user denying access).
Chrome is not throwing either.